src/base/preferences.cpp

```diff
@@ -582,28 +581,16 @@ void Preferences::setWebUiUsername(const QString &username)
     setValue("Preferences/WebUI/Username", username);
 }
 
-QString Preferences::getWebUiPassword() const
+QByteArray Preferences::getWebUIPassword() const
 {
-    QString passHa1 = value("Preferences/WebUI/Password_ha1").toString();
-    if (passHa1.isEmpty()) {
-        QCryptographicHash md5(QCryptographicHash::Md5);
-        md5.addData("adminadmin");
-        passHa1 = md5.result().toHex();
-    }
-    return passHa1;
+    // default: adminadmin
+    const QByteArray defaultValue = "ARQ77eY1NUZaQsuDHbIMCA==:0WMRkYTUWVT9wVvdDtHAjU9b3b7uB8NR1Gur2hmQCvCDpm39Q+PsJRJPaCU51dEiz+dTzh8qbPsL8WkFljQYFQ==";
+    return value("Preferences/WebUI/Password_PBKDF2", defaultValue).toByteArray();
```

Comment:
What's the reason to use such verbose param name? Why not just "Preferences/WebUI/Password"?
The same question is for other password.

Reply:
>What's the reason to use such verbose param name? 

It helps to differentiate different password hashing methods in case we need to change the method again in the future, of course we can use "Preferences/WebUI/Password" or "PasswordV2", "PasswordV3" and so on but why not just use its method name instead?

Reply:
>why not just use its method name instead?

To keep it from being declassified, since you care so much about safety.

Reply:
>To keep it from being declassified, since you care so much about safety.

Ha, but hiding its name in this case won't help much, we are open source :)
Let me know if you insist on this.

Reply:
> if you insist on this.

No.
But wouldn't it be good to clean up old keys from config file? Otherwise persistent users accumulate obsolete keys that are no longer in use. We don't change it very often though... So I don't insist.

Reply:
>But wouldn't it be good to clean up old keys from config file? 

Yes but it might break users who downgrades (and the same applies to all config keys), so I'm not sure.